### PR TITLE
Release notes 2.1.2

### PR DESCRIPTION
**Pull request by nathan-px**
2.1.2 release notes.

### CODE REVIEW

**Comment by nathan-px** [comment]
I see two problems: no recommendations on 8635, and nothing on 9056. 

I don't have content for either, so I can try to ping the SMEs again, but am prepared to remove them.

File: content/reference/release-notes/_index.md [hunk 1]
```diff
@@ -6,6 +6,205 @@ keywords: portworx, release notes
...
+_Customer Impact:_  Previously, if you did not specify an enforcement type in `VolumePlacementStrategy`, PVC creation failed. With 2.1.2, Portworx will default to `encorement: required` if you do not specify an enforcement type.
+
+
+_Recommendations:_ Portworx recommends that you upgrade to 2.1.2.
+
+**PWX-8959** - The Snapshot API does not return a typed error if a snapshot already exists.
+
+_Customer Impact:_ With 2.1.2, this error message is improved. If you try to create a new snapshot where one already exists, a clearly typed error is now returned.
+
+_Recommendations:_ Portworx recommends upgrading to 2.1.2.
```
Comment (harsh-px) [inline content/reference/release-notes/_index.md]: Not a big fan of the `_Recommendations:_ Portworx recommends upgrading to 2.1.2.` format.

Seems redundant and doesn't provide value.

Comment (nathan-px) [reply content/reference/release-notes/_index.md]: I agree, Venkat expressed a desire for us not to change the format abruptly for this release, so I went with what I saw was already there. Additionally, Vilas said that there was no workaround information to add to the recommendations, and so now we have nothing to add. 

I will strike them from the doc.

**Comment by nathan-px** [comment]
@piyush-nimbalkar @vilasd @venkatpx I added the Azure errata. Let me know if it looks correct

File: content/reference/release-notes/_index.md [hunk 2]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+_Recommendations:_ Upgrade to 2.1.2 to support rolling upgrades in cloud auto-scaling environments. If you are unable to upgrade to 2.1.2, you can work with Portworx support to get temporary licenses that increase the node count until an upgrade to 2.1.2 can be planned.
+
+**PWX-9042** - Secrets can be overwritten with certain commands.
+
+_Customer Impact:_ with 2.1.2, It is no longer possible to unintentionally overwrite secrets using a combination of the `pxctl secrets set-cluster-key` and `pxctl generate secret` commands.
+
+**PWX-8953** -   (Consul) KVDB does not pass the CAFile to the Consul client.
+
+_Customer Impact:_ With 2.1.2, CAFile is now properly sent to the Consul client.
```
Comment (vilasd) [inline content/reference/release-notes/_index.md]: What is CAFile ??? Should we put that in the description ?

File: content/reference/release-notes/_index.md [hunk 3]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+**PWX-8966** - (ASG) Limit the maximum number of drives you can attach to a node.
+
+_Customer Impact:_  Previously, Attempting to attach too many drives results resulted in errors. With 2.1.2, Portworx introduces a maximum limit of 12 drives per node, and will respond to attempts to add more than 12 drives with the following error: `cannot provide more than 12 number of drives per node`.
+
+**PWX-7374** - AWS SC1 and ST1 EBS volumes are unsupported.
+
+_Customer Impact:_ It was previously not possible to add SC1 or ST1 EBS volumes using the `pxctl sv drive add` command. With 2.1.2, Portworx now supports SC1 and ST1 EBS drive types.
+
+**PWX-8792** - Add backoffs to AWS CloudDrive APIs.
```
Comment (vilasd) [inline content/reference/release-notes/_index.md]: Instead of APIs  can we say something else ?

File: content/reference/release-notes/_index.md [hunk 4]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+_Customer Impact:_ With 2.1.2, Portworx now slows calls to the AWS Cloud Drive API at increasingly long intervals if it encounters resource limit errors.
+
+**PWX-8701** - The internal KVDB should use DNS names for peer urls.
+
+_Customer Impact:_ The internal KVDB tracks peer URLs as potentially ephemeral IP addresses. If the entire cluster becomes unavailable in an outage, Portworx may be unable to reconnect. With 2.1.2, internal KVDB now keeps track of nodes using a DNS, and can therefore reconnect in the event of a total cluster outage.
+
+**PWX-8904** - In certain situations, nodes may get stuck on startup.
```
Comment (vilasd) [inline content/reference/release-notes/_index.md]: Its not nodes its portworx get stuck on startup . I added this to the title on confluence page  "Introduce timeout on storage requests to avoid possible hung situations on node start"

Comment (vilasd) [reply content/reference/release-notes/_index.md]: we can make this portworx start

File: content/reference/release-notes/_index.md [hunk 5]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+_Customer Impact:_ Previously, nodes may have failed to start, displaying no active I/O operations. With 2.1.2, storage requests now timeout to avoid possible hung situations on node start.
+
+**PWX-8606** - PVC creation fails if no enforcement type is specified in `VolumePlacementStrategy`.
+
+_Customer Impact:_  Previously, if you did not specify an enforcement type in `VolumePlacementStrategy`, PVC creation failed. With 2.1.2, Portworx will default to `encorement: required` if you do not specify an enforcement type.
```
Comment (vilasd) [inline content/reference/release-notes/_index.md]: Spelling mistake "encorement" --> "enforcement"

File: content/reference/release-notes/_index.md [hunk 6]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+_Customer Impact:_ With 2.1.2, this error message is improved. If you try to create a new snapshot where one already exists, a clearly typed error is now returned.
+
+**PWX-9126** - The `nodiscard` option is impacted by volume resize.
+
+_Customer Impact:_ Previously, resizing a volume sometimes reset a volume’s `nodiscard` option configuration. With 2.1.2, this has been fixed.
+
+**PWX-8690** - Wipe and upgrade scripts fail on Kubernetes 1.14.
+
+_Customer Impact:_ Due to a feature deprecation with Kubernetes 1.14, wipe and upgrade scripts did not work. With 2.1.2, The `px-wipe` command now correctly removes pods on Kubernetes 1.14.
```
Comment (vilasd) [inline content/reference/release-notes/_index.md]: remove portworx pods not pods

File: content/reference/release-notes/_index.md [hunk 7]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+**PWX-9175** - Portworx fails to detect the device path on certain types of AWS ec2 instances.
+
+_Customer Impact:_ With 2.1.2, Portworx no longer fails to detect the device path on certain types of AWS ec2 instances and operating system combinations.
+
+**PWX-7851** - Storage nodes could removed by the Kubernetes autoscaler.
```
Comment (vilasd) [inline content/reference/release-notes/_index.md]: Storage nodes could removed ===> Storage nodes could be removed

File: content/reference/release-notes/_index.md [hunk 8]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+_Customer Impact:_ With 2.1.2, Portworx pods are annotated to prevent node removal by the Kubernetes autoscaler.
+
+**PWX-7493** - Portworx selects undesired network interfaces during autodetection.
+
+_Customer Impact:_ With 2.1.2, Portworx will avoid selecting the undesired network interfaces during configuration.
+
+**PWX-9053** - Portworx fails to wipe MDRAID devices.
```
Comment (vilasd) [inline content/reference/release-notes/_index.md]: pxctl node wipe command fails to wipe MDRAID devices.

File: content/reference/release-notes/_index.md [hunk 9]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+_Customer Impact:_ With 2.1.2, Portworx now detects MDRAID partitions when installed with the `-A` option.
+
+**PWX-8938** - Add “sync” and “noac” sharedv4 mount options.
+
+_Customer Impact:_ With 2.1.2,  “sync” and “noac” sharedv4 mount options are now available.
+
+**PWX-8893** - Add support for `max_storage_nodes_per_zone` on AKS clusters on Microsoft Azure.
+
+_Customer Impact:_ With 2.1.2, AKS clusters deployed across availability sets on Microsoft Azure support  `max_storage_nodes_per_zone` to limit storage nodes on availability sets.
```
Comment (vilasd) [inline content/reference/release-notes/_index.md]: reword the impact as it is including Microsoft azure support which is confusing. With 2.1.2, portworx supports `max_storage_nodes_per_zone` to limit storage nodes on availability sets on AKS clusters deployed across availability sets

File: content/reference/release-notes/_index.md [hunk 10]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+**PWX-9263** - Provisioning an XFS volume creates an EXT4 formatted volume.
```
Comment (vilasd) [inline content/reference/release-notes/_index.md]: Should we reword this as it might make people think what are they running underneath  ? I think old bug title "Volumes can now be created with XFS filesystem without needing to provide the "force_unsupported_fs_type" option." was good.

File: content/reference/release-notes/_index.md (hunk 2 above)
Comment (nathan-px) [reply content/reference/release-notes/_index.md]: CA Certificate file

File: content/reference/release-notes/_index.md (hunk 3 above)
Comment (nathan-px) [reply content/reference/release-notes/_index.md]: API calls

File: content/reference/release-notes/_index.md (hunk 4 above)
Comment (nathan-px) [reply content/reference/release-notes/_index.md]: Change title to this: "Introduce timeout on storage requests to avoid possible hung situations on Portworx start"

change "nodes" to "Portworx"

File: content/reference/release-notes/_index.md (hunk 6 above)
Comment (nathan-px) [reply content/reference/release-notes/_index.md]: change pods to Portworx pods

File: content/reference/release-notes/_index.md (hunk 8 above)
Comment (nathan-px) [reply content/reference/release-notes/_index.md]: The `pxctl sv node wipe` command fails to wipe MDRAID devices.

File: content/reference/release-notes/_index.md (hunk 9 above)
Comment (nathan-px) [reply content/reference/release-notes/_index.md]: With 2.1.2, portworx supports the `max_storage_nodes_per_zone` parameter, allowing it to limit storage nodes on AKS availability sets.

File: content/reference/release-notes/_index.md (hunk 10 above)
Comment (nathan-px) [reply content/reference/release-notes/_index.md]: Volumes can now be created with XFS filesystem without needing to provide the "force_unsupported_fs_type" option.

File: content/reference/release-notes/_index.md [hunk 11]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+_Customer Impact:_ with 2.1.2, It is no longer possible to unintentionally overwrite secrets using a combination of the `pxctl secrets set-cluster-key` and `pxctl generate secret` commands.
+
+**PWX-8953** -   (Consul) KVDB does not pass the CA certificate file to the Consul client.
+
+_Customer Impact:_ With 2.1.2, CAFile is now properly sent to the Consul client.
+
+**PWX-8966** - (ASG) Limit the maximum number of drives you can attach to a node.
+
+_Customer Impact:_  Previously, Attempting to attach too many drives results resulted in errors. With 2.1.2, Portworx introduces a maximum limit of 12 drives per node, and will respond to attempts to add more than 12 drives with the following error: `cannot provide more than 12 number of drives per node`.
```
Comment (harsh-px) [inline content/reference/release-notes/_index.md]: Attempting -> `attempting`

File: content/reference/release-notes/_index.md [hunk 12]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+**PWX-9476** - OCI-Monitor should not restart the Portworx service unless required.
+
+_Customer Impact:_ Previously, if the OCI-Monitor restarted Portworx any time it detected a change in the configuration file. WIth 2.1.2, the OCI-Monitor will only restart Portworx if it’s necessary.
+
+### Errata
+
+**PWX-9473** - Portworx fails to attach a cloud drive to an Azure AKS scale set VM.
+
+_Customer Impact:_ Portworx fails to attach a cloud drive with the following error messages: `Failed to detach disk...`, and `Failed to attach drive...`
```
Comment (harsh-px) [inline content/reference/release-notes/_index.md]: Replace `Customer Impact` to `User Impact`

File: content/reference/release-notes/_index.md [hunk 13]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
 
+## 2.1.2
+
+July 24, 2019
+
+### Key Features:
+
+1. [Cloud drive support for Microsoft Azure](/portworx-install-with-kubernetes/cloud/azure/aks/)
+2. Red Hat OpenShift 4.1 Operator Support
+3. Enhanced Volume placement strategy management
```
Comment (harsh-px) [inline content/reference/release-notes/_index.md]: `Volume placement strategy management` => `Volume placement strategies for advanced volume provisioning rules`

File: content/reference/release-notes/_index.md [hunk 14]
```diff
@@ -6,6 +6,146 @@ keywords: portworx, release notes
...
+
+### Enhancements:
+
+**PWX-8635** - Add support for the CRI-O container runtime.
+
+_User Impact:_ Portworx has added support for the CRI-O container runtime, with the some log file limitations:
+
+  * The progress bar while downloading images is not available.
+  * Progresss information while installing Portworx binaries is not available.
```
Comment (harsh-px) [inline content/reference/release-notes/_index.md]: Progresss typo
